fix compile errors and tests logic

zoedberg opened the pull request:
Ref. https://github.com/RGB-Tools/rgb-lib/issues/22

Lifetime compile errors were mainly due to a logic error in test construction. By sharing the same `Wallet` instance across tests it was impossible for them to execute successfully. The backend should share the wallet instance, while each test should create a new one. This is because the backend needs the wallet to be the same across subsequent API calls while each test represents a whole new backend scenario, so each one needs to create a new wallet and each subsequent API call in the test needs to use the same wallet to test the desired operation in isolation.

Code now compiles and tests pass. To achieve this, I've removed `WalletTestContext` and all the related code that was incorrectly trying to share the same wallet instance across different tests. Now each test creates its backend, with its empty `ShiroWallet` instance, calls the `/wallet` API to create a new rgb-lib `Wallet` with the desired mnemonic and then calls the API which is being tested.

On the backend side I've wrapped the `Mutex` with an `Arc` because reading some actix_web guides I've discovered it's important in order to safely share the instance across threads.

@monaka please let me know if something needs clarification

monaka commented:
As I wrote in Telegram, I suppose an instance of `Wallet` must be shared between tests.
Because a new thread will be spawned in `go_online()` and it won't stop even if the instance was released.

zoedberg commented:
Sharing the same `Wallet` across multiple tests is wrong because each test recreates a new backend. Each backend needs to use its own wallet, internally sharing the instance (once created), so each test needs to create a different wallet.

It's true that `go_online` spawns some threads that cannot be stopped (except by killing main process, which is undesirable) but this doesn't imply that tests should share the same wallet.

Confusion may arise from the fact that only one wallet can go online on the same data directory (where the spawned threads hold locks on files). This stems from the fact that wallets created with the same xpub will use the same data directory. But this can be solved by simply using different keys (hence data directories) on each test.

I've updated this PR with a commit that shows how to call `go_online` on multiple wallets without any change to rgb-lib. To summarize, this is what I've done:
- added the API to go online and a new test (`test_put_again`) that shows how you should create different `WalletParams` for each test (using `generate_keys`) so that multiple tests don't end up instantiating multiple wallets on the same data directory
- removed the `Arc` wrapping `ShiroWallet` that I was mentioning on the PR description, because I've discovered that, in recent actix_web releases, `web::Data` already wraps its content with an `Arc`
- in `ShiroWallet`, removed the `WalletState` enum and added a dedicated field for the `Wallet` for the sake of simplicity
- changed the new address API so now it doesn't automatically create the wallet (again for simplicity)

monaka commented:
@zoedberg Thank you for your contribution.
I cherry-picked your commits by hand as your branch and my branch conflicts.
They'll be merged to my main branch soon.
I close this PR as already applied.
